runtime/lua/vim/lsp/util.lua

```diff
@@ -610,7 +610,7 @@ end
 ---@return string parsed snippet
 function M.parse_snippet(input)
   local ok, parsed = pcall(function()
-    return tostring(snippet.parse(input))
+    return tostring(snippet.parse(input).data.children)
```

Comment:
Looking at how `parse_snippet` is currently used in https://github.com/MariaSolOs/neovim/blob/5f7c18e554ad8126176371b1593eddfafa10049c/runtime/lua/vim/lsp/util.lua#L633-L650

I don't think this is safe to do. E.g. `lua =vim.lsp.util.parse_snippet("${1:foobar}")` on master returns `foobar`, now it would return `table: 0x...`.

I'm not sure how to do this in a non-disruptive way. We could change `get_completion_word` to not use this function anymore - without snippet support opt-in in the capabilities users shouldn't get results anyways.

But not sure what to do about direct `parse_snippet` users. As mentioned in the other comment. I think this function needs to 🔥

If a quick github code search shows no other usages, maybe we're daring a ☢️ option and kick it immediately ?

Reply:
It's [used by nvim-cmp](https://github.com/hrsh7th/nvim-cmp/blob/5dce1b778b85c717f6614e3f4da45e9f19f54435/lua/cmp/view/ghost_text_view.lua#L75), unfortunately (@hrsh7th), as well as a number of dotfiles (there seems to be a copy-pasted snippet that throws out multiline snippets).

I'm OK with breaking user config on master in this case, but maybe we can coordinate with nvim-cmp so that they have an adaptation in place when we merge this?

Alternative is to not merge this "partial" PR and combine it with the API one, so we can give this the `stylize_markdown` treatment and tell people "just use `vim.snippet.parse()` (or whatever we decide on) instead". That's probably the easiest for downstream.

Reply:
Or is it possible to translate the new result into the old form with a simple shim we can insert now and remove before 0.10?

Reply:
I will just move all of the current parse snippet logic inside `vim.lsp.util.parse_snippet` (so basically encapsulate the shim there). We can deprecate it later once we have the API.

Reply:
B-but your diffstats... 😭

Reply:
😭